segmentstore/server/src/main/java/io/pravega/segmentstore/server/reading/CacheIndexEntry.java

```diff
@@ -10,34 +10,56 @@
 package io.pravega.segmentstore.server.reading;
 
 import com.google.common.base.Preconditions;
+import javax.annotation.concurrent.GuardedBy;
+import lombok.Getter;
 
 /**
  * Represents a ReadIndexEntry that points to an entry in the Cache.
  */
 public class CacheIndexEntry extends ReadIndexEntry {
-    private final int length;
+    @Getter
+    private final int cacheAddress;
+    @GuardedBy("this")
+    private int length;
 
     /**
      * Creates a new instance of the ReadIndexEntry class.
      *
      * @param streamSegmentOffset The StreamSegment offset for this entry.
      * @param length              The Length of this entry.
+     * @param cacheAddress        The address of this Index Entry in the CacheStorage.
      * @throws IllegalArgumentException if the offset is a negative number.
      * @throws IllegalArgumentException if the length is a negative number.
      */
-    CacheIndexEntry(long streamSegmentOffset, int length) {
+    CacheIndexEntry(long streamSegmentOffset, int length, int cacheAddress) {
         super(streamSegmentOffset);
         Preconditions.checkArgument(length >= 0, "length", "length must be a non-negative number.");
         this.length = length;
+        this.cacheAddress = cacheAddress;
     }
 
     @Override
-    long getLength() {
+    synchronized long getLength() {
         return this.length;
     }
 
+    /**
+     * Increases the length by the given amount.
+     *
+     * @param delta The amount to increase by.
+     */
+    synchronized void increaseLength(int delta) {
+        Preconditions.checkArgument(delta >= 0, "delta must be a non-negative number.");
+        this.length += delta;
+    }
+
     @Override
     boolean isDataEntry() {
         return true;
     }
+
+    @Override
+    public synchronized String toString() {
```

Comment:
Why does this need to be synchronized? `cacheAddress` has no `@GuardedBy` annotation.

Reply:
No it doesn't. I don't know how it ended up there.